### PR TITLE
API code reorganization

### LINKED ISSUE #510: Split up API and serializer code by object
## Problem

Currently, our API code and serializer code are in one file each (`views/api.py` and `serializers.py`). These files are getting long.

## Proposed solution

We should split this up per-object (e.g. `views/api/tables.py` and `serializers/tables.py`). It'll be easier to find code and it will be consistent with how we organize our code in other places.

### PR DESCRIPTION
**Pull request by kgodey**
Fixes #510 

Moves API code into a single folder for easier discoverability and splits it up into per-object files. This is all moving code around, there were no logic changes made.

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `master` branch of the repository
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by eito-fis — COMMENTED** [review]
Mostly just skimmed the new structure, but I think it looks great :+1:

**Comment by powellc** [comment]
@kgodey I think this will make more sense once I see it in the file tree, but reading through the PR all the changes look great. I like the simplification of imports and the restructuring of viewsets.

**Review by powellc — APPROVED** [review]

File: mathesar/api/serializers/columns.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
+from rest_framework import serializers
+from rest_framework.exceptions import ValidationError
+from rest_framework.fields import empty
+from rest_framework.settings import api_settings
+
+
+class InputValueField(serializers.CharField):
+    """
+    Takes in an arbitrary value. Use to emulate our column and record creation and
+    update endpoints, which handle arbitrary data pulled from request.data
```
Comment (mathemancer) [inline mathesar/api/serializers/columns.py:10]: Is this comment accurate?  I note that this is only used in this column serializer file.  If it's intended to be used for records as well, consider moving it to a `util` file.

File: mathesar/urls.py [hunk 2]
```diff
@@ -1,27 +1,34 @@
...
 
-from mathesar.views import api, frontend
+from mathesar import views
+from mathesar.api.viewsets.columns import ColumnViewSet
+from mathesar.api.viewsets.constraints import ConstraintViewSet
+from mathesar.api.viewsets.data_files import DataFileViewSet
+from mathesar.api.viewsets.databases import DatabaseViewSet
+from mathesar.api.viewsets.records import RecordViewSet
+from mathesar.api.viewsets.schemas import SchemaViewSet
+from mathesar.api.viewsets.tables import TableViewSet
```
Comment (mathemancer) [inline mathesar/urls.py:11]: It might be worth adding these to the `api` namespace for easier importing using the appropriate `__init__.py`.

**Review by mathemancer — APPROVED** [review]
Looks good overall.  I made a couple of comments about optional changes.  Please feel free to make any or all or none of them as you see fit.  I'll re-review if you want to change anything; otherwise go ahead and merge.

File: mathesar/api/serializers/columns.py (hunk 1 above)
Comment (kgodey) [reply mathesar/api/serializers/columns.py:10]: It is not used elsewhere. I copied the comment as-is, I'll fix the comment in a future PR.

File: mathesar/urls.py (hunk 2 above)
Comment (kgodey) [reply mathesar/urls.py:11]: Good idea.

**Pull request merged**